nvflare/app_common/utils/fl_model_utils.py

```diff
@@ -93,9 +92,7 @@ def to_shareable(fl_model: FLModel, params_converter: Optional[ParamsConverter]
         return shareable
 
     @staticmethod
-    def from_shareable(
-        shareable: Shareable, params_converter: Optional[ParamsConverter] = None, fl_ctx: Optional[FLContext] = None
-    ) -> FLModel:
+    def from_shareable(shareable: Shareable, params_converter: Optional[ParamsConverter] = None) -> FLModel:
```

Comment:
I'm not sure about this. There is a lot of info in fl_ctx that could be needed in FLModel in the future. We may not need much from fl_ctx right now, but that doesn't mean we'll never need it. I think we should keep fi_ctx here.

Reply:
I added this "FlareAgentWithFLModel", in this class I use these methods (from_shareable, to_shareable) to convert the shareable to FLModel and vice versa.

Since the FlareAgentWithFLModel is running on 3rd party, it does not have FLContext though.

I do have a case that need something from fl_ctx in the case of LauncherExecutor, what I did is that I set the shareable header first before sending it via pipe to the 3rd party.

Reply:
You can pass None for fl_ctx in the case that you don't have it. Just make sure that when you use prop from fl_ctx, check for None first.

Reply:
OK, let me keep the signature so if we want more stuff from fl_ctx we can do that.